keras/src/losses/loss.py

```diff
@@ -25,6 +25,12 @@ class Loss(KerasSaveable):
             `"float32"` unless set to different value
             (via `keras.backend.set_floatx()`). If a `keras.DTypePolicy` is
             provided, then the `compute_dtype` will be utilized.
+        normalize_by_sample_weight: Whether to normalize the loss value by the
```

Comment:
This seems like it should be a reduction type instead?

Reply:
That sounds more reasonable. A new option called `"mean_with_sample_weight"` has been introduced.
In addition, I have updated the the `reduction` docstrings for all losses.

```python
        reduction: Type of reduction to apply to the loss. In almost all cases
            this should be `"sum_over_batch_size"`. Supported options are
            `"sum"`, `"sum_over_batch_size"`, `"mean"`,
            `"mean_with_sample_weight"` or `None`. `"sum"` sums the loss,
            `"sum_over_batch_size"` and `"mean"` sum the loss and divide by the
            sample size, and `"mean_with_sample_weight"` sums the loss and
            divides by the sum of the sample weights. `"none"` and `None`
            perform no aggregation. Defaults to `"sum_over_batch_size"`.
``` 

A minor bug in numpy's `divide_no_nan` has also been resolved.